### PR TITLE
refactor(server/main): remove getCoralInventory

### PR DESCRIPTION
**Pull request by googleoblivion**
## Description
availableCoral table is malformed and is unnecessary. Merges the function into the main event.

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: server/main.lua [hunk 1]
```diff
@@ -8,48 +8,38 @@ local pickedUpCoralIndexes = {}
...
+        local coral = config.coralTypes[i]
+        local count = exports.ox_inventory:GetItemCount(src, coral.item)
 
-    for coral, amount in pairs(availableCoral) do
-        local price = amount * coral.price
-        local reward = getItemPrice(amount, price)
-        exports.ox_inventory:RemoveItem(src, coral.item, amount)
-        player.Functions.AddMoney('cash', math.ceil(reward / amount), 'sold-coral')
+        if count and count > 0 then
+            corals = corals - 1
```
Comment (Manason) [inline server/main.lua]:
```suggestion
            corals -= 1
```

Comment (Manason) [inline server/main.lua]: Does modifying this value affect the for loop iteration? Or is the number of iterations set once and then the underlying variable change won't affect it?

**Review by Manason — APPROVED** [review]

File: server/main.lua (hunk 1 above)
Comment (googleoblivion) [reply server/main.lua]: @Manason can you clarify what you mean here? The corals -= 1 affecting for loop? corals is just the counter for the notify under it.

Comment (Manason) [reply server/main.lua]: The for loop is iterating from 1 to the value of corals, but then while executing, corals is being decremented. I'm wondering if this cuts the number of iterations short.

**Review by Manason — CHANGES_REQUESTED** [review]
Instead of giving the money piecemeal, it may be better to track how much money should be given, and give it once. This also would make it so corals isn't needed to be tracked separately

**Review by Manason — APPROVED** [review]

**Review by mafewtm — APPROVED** [review]

**Pull request merged**